packages/website/tests/playground.spec.ts

```diff
@@ -46,6 +46,19 @@ test.describe('Playground', () => {
     await expect(page.getByText('let value: string[];')).toBeVisible();
     await expect(page.getByText('All is ok!')).toBeVisible();
   });
+
+  test('AST Viewer', async ({ page }) => {
+    // 1. Type some valid code in the playground
+    await writeInEditor(page, 'let value: Array<string>;');
+    // 2. Enable AST viewer
+    await page
+      .getByRole('combobox', { name: 'AST Viewer' })
+      .selectOption({ label: 'ESTree' });
+    // 3. Validate variable declaration block exists in AST viewer
+    await expect(
+      page.getByRole('link', { name: 'VariableDeclaration' }),
+    ).toBeVisible();
+  });
 });
 
 async function writeInEditor(page: Page, text: string): Promise<void> {
```

Comment:
Threading https://github.com/typescript-eslint/typescript-eslint/pull/6275#issuecomment-1365289946: hmm, yeah, I see what you mean. When I run it locally, it looks like Monaco is wiping out the typed text:

1. The test selects & deletes all text
2. The test types in new text
3. ...some hundreds of milliseconds later...
4. Monaco deletes all the newly added text

My first instinct for a workaround is to wait until Monaco is done processing whatever it needs to do. Vaguely:

1. Type some bogus text into the editor
2. Wait until the text dissapears
3. ...then do the rest of the text

I prototyped this locally and it worked out well. What do you think?

Reply:
Hi @JoshuaKGoldberg, Workaround looks good. Meanwhile, I was checking another workaround by looking at the previous test case. I tried out something like, opening the lint editor and going back to the code editor again. Then the typed text persisted without wiping out. It worked locally. Modified the current test based on it.

Reply:
> My first instinct for a workaround is to wait until Monaco is done processing whatever it needs to do. Vaguely:

workaround should not be needed, someone reported similar issue in browser, we should fix this bug

Reply:
this issue should be fixed now (#6513), as long as you focus editor content is not going to be cleared anymore :)

Reply:
to focus editor and wait for it you have to do something like this:

```
  await page.isVisible('.monaco-editor .monaco-mouse-cursor-text');
  const monacoEditor = page
    .locator('.monaco-editor .monaco-mouse-cursor-text')
    .nth(0);
  await monacoEditor.focus();
```

and as for this test, you can do something like this:

```
    // 1. Enable AST viewer
    await page
      .getByRole('combobox', { name: 'AST Viewer' })
      .selectOption({ label: 'ESTree' });

    // 2. Type some valid code in the playground
    await writeInEditor(page, 'let value: Array<string>;');

    // 3. Wait for the AST viewer to update
    await page.waitForEvent(
      'console',
      message => message.text() === '[Editor] linting triggered',
    );

    // 4. Validate variable declaration block exists in AST viewer
    await expect(
      page.getByRole('link', { name: 'VariableDeclaration' }),
    ).toBeVisible();
```